docs/source/features/parallelisms.rst

```diff
@@ -41,54 +41,83 @@ Enable Data Parallelism
-===========================  =========  ==================================================================================================================================
-
-See the keyword arguments in `Apex DistributedFusedAdam <https://github.com/NVIDIA/apex/blob/master/apex/contrib/optimizers/distributed_fused_adam.py>`_ and `NeMo MegatronDistributedFusedAdam <https://github.com/NVIDIA/NeMo/blob/main/nemo/core/optim/distributed_adam.py>`_ for a full list of distributed optimizer options.
-
-Implement Data Parallelism
-~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-DDP in NeMo Framework uses either PyTorch
-`DistributedDataParallel <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`_
-(default) or a custom implementation (if custom multi-precision
-training is enabled with ``megatron_amp_O2``).
-
-The distributed optimizer in NeMo Framework is built on top of
-`DistributedFusedAdam <https://github.com/NVIDIA/apex/blob/master/apex/contrib/optimizers/distributed_fused_adam.py>`_
-from Apex.
-
-Fully-Shared Data Parallelism
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-NeMo Framework supports Fully-Sharded Data Parallelism (FSDP), which shards parameter gradients and low-precision parameters for computation. This is in addition to the model states that the distributed optimizer shards, including optimizer states and high-precision parameters.
-Since FSDP shards the entire model states, it ensures linear model state memory savings with increasing DP size.
-FSDP is preferred for LLM training with unbalanced workloads between pipeline stages (or Transformer layers) or with a large vocabulary size, where pipelining would cause significant computation bubbles due to workload imbalance.
-Additionally, FSDP eliminates the need to search for performance-optimal mappings with 3D parallelism (TP/PP/DP) because it operates within a single parallelization domain.
-
-
-NeMo Framework uses `PyTorch's FSDP interface <https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html>`_ to shard LLM model states, flattening the parameters of each transformer layer and partitioning them across data-parallel GPUs.
-FSDP introduces collective operations across data-parallel GPUs, including all-gather for parameter computation and reduce-scatter for parameter gradients.
-The all-gather operation occurs during both the network forward and back-propagation phases, while the gradient reduce-scatter operation happens only during back-propagation.
-These FSDP communications are overlapped with transformer layer computations.
-
-Setting ``fsdp=true`` enables FSDP.
-The mixed precision recipe can be set by ``precision`` knob, which determines both the computation and communication precisions.
-Also, one can use ``grad_reduce_dtype`` to override the gradient reduction precision specifically.
+Currently, NeMo Framework supports optimizer distribution only for Megatron-Core adam distributed optimizer.
```

Comment:
revise

Currently, NeMo Framework supports optimizer distribution only for the Megatron Core Adam distributed optimizer.